src/slot.c

```diff
@@ -122,7 +123,11 @@ int parse_slots()
             continue;
         }
 
-        node_list_add(n);
+        size_t num = MIN(n->index, MAX_NODE_LIST - tmp_nodes_len);
+        if (num > 0) {
+            memcpy(tmp_nodes + tmp_nodes_len, n->nodes, num * sizeof(struct address));
+            tmp_nodes_len += num;
+        }
```

Comment:
It would be great to wrap these into a function, it's an append operation after all.